extensions/autodoc_field/__init__.py

```diff
@@ -0,0 +1,115 @@
+        elif field.type in ('one2many', 'many2many'):
+            annotation[attrname] = Sequence[odoo.fields.Command]
+        elif field.type in ('selection', 'reference', 'char', 'text', 'html'):
+            annotation[attrname] = str
+        elif field.type == 'boolean':
+            annotation[attrname] = bool
+        elif field.type in ('float', 'monetary'):
+            annotation[attrname] = float
+        elif field.type == 'integer':
+            annotation[attrname] = int
+        elif field.type == 'date':
+            annotation[attrname] = datetime.date
+        elif field.type == 'datetime':
+            annotation[attrname] = datetime.datetime
+
+    def add_content(self, more_content):
+        source_name = self.get_sourcename()
+        field = self.object
+        if field.required:
+            self.add_line(f"| **Required**", source_name)
+        self.add_line(f"| **Name:** {field.string}", source_name)
+        if field.readonly:
+            self.add_line(f"| **Read-only:** this field is not supposed to/cannot be set manually", source_name)
+        if not field.store:
+            self.add_line(f"| **Not stored:** this field is there only for technical reasons", source_name)
+        if field.type == 'selection':
+            if isinstance(field.selection, (list, tuple)):
+                self.add_line(f"| **Possible values:**", source_name)
+                for tech, nice in field.selection:
+                    self.add_line(f"|     `{tech}`: {nice}", source_name)
+        if field.type in ('many2one', 'one2many', 'many2many'):
+            comodel_name = field.comodel_name
+            string = f"| **Comodel:** :ref:`{comodel_name} <model-{comodel_name.replace('.', '-')}>`."
+            self.add_line(string, source_name)
+            reference = self.config.model_reference.get(comodel_name)
+            if reference:
+                self.add_line(f"| **Possible values:** `{reference} <{self.config.ultimate_replacements['{GITHUB_PATH}']}/{reference}>`__.", source_name)
+        if field.default:
+            self.add_line(f"| **Default:** {field.default(odoo.models.Model)}", source_name)
+        if field.help:
+            for line in field.help.split('\n'):
+                self.add_line(f"| {line}", source_name)
+        self.add_line('', source_name)
+        super().add_content(more_content)
+
+    def get_doc(self, encoding=None, ignore=None):
+        # only read docstring of field instance, do not fallback on field class
+        field = self.object
+        field.__doc__ = field.__dict__.get('__doc__', "")
+        res = super().get_doc(encoding, ignore)
+        return res
+
+def disable_warn_missing_reference(app, domain, node):
+    if (domain and domain.name != 'std') or node['reftype'] != 'ref':
+        return None
+    else:
+        target = node['reftarget']
+        if target.startswith('model-'):
+            node['reftype'] = 'odoo_missing_ref'
+            return True
```

Comment:
```suggestion
    if (not domain or domain.name == 'std') and node['reftype'] == 'ref':
        target = node['reftarget']
        if target.startswith('model-'):
            node['reftype'] = 'odoo_missing_ref'
            return True
```
Seems simpler and to work the same? Though the first part of teh guard seems a bit weird.

Also a docstring to more clearly explain the purpose of this handler would be nice.